micrometer-core/src/test/java/io/micrometer/core/instrument/binder/jvm/JvmGcMetricsTest.java

```diff
@@ -65,13 +65,18 @@ void noJvmImplementationSpecificApiSignatures() {
     }
 
     @Test
-    void metersAreBound() {
-        assertThat(registry.find("jvm.gc.live.data.size").gauge()).isNotNull();
-        assertThat(registry.find("jvm.gc.memory.allocated").counter()).isNotNull();
-        assertThat(registry.find("jvm.gc.max.data.size").gauge().value()).isGreaterThan(0);
+    void gcMetricsAvailableAfterGc() {
+        System.gc();
+        await().timeout(200, TimeUnit.MILLISECONDS).alias("NotificationListener takes time after GC")
+                .untilAsserted(() ->
+                        assertThat(registry.find("jvm.gc.live.data.size").gauge().value()).isPositive());
+        assertThat(registry.find("jvm.gc.memory.allocated").counter().count()).isPositive();
```

Comment:
This metric is still missing for `optavgpause` and `optthruput` due to the allocation pool name not being set. We don't have a CI job run with OpenJ9 because we're still waiting for CI images (possibly https://github.com/CircleCI-Public/cimg-openjdk/issues/78). I think we can fix the missing allocated counter separately from this pull request because it is still an improvement to get rid of the NPE which allows updating of the live and max data size gauges.